src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCache.cs

```diff
@@ -669,8 +660,12 @@ private static void ValidateCacheKey(object key)
         /// </summary>
         private sealed class CoherentState
         {
+#if NETCOREAPP
```

Comment:
We can consistently use the same constant https://github.com/dotnet/runtime/pull/101693

```suggestion
#if NET
```